embark.el

```diff
@@ -942,60 +961,80 @@ display actions and parameters are available."
                 (string-match-p x (symbol-name cmd))))
             embark-verbose-indicator-excluded-commands))
 
-(defun embark--verbose-indicator-update (keymap target other-targets)
+(defun embark--verbose-indicator-insert-target (kind target)
```

Comment:
Instead of inserters use pure functions retuning strings as you already suggested? Seems cleaner to me.